homeassistant/components/shelly/__init__.py

```diff
@@ -289,13 +294,17 @@ def _async_device_online(_: Any, update_type: RpcUpdateType) -> None:
             data[CONF_SLEEP_PERIOD] = get_rpc_device_wakeup_period(device.status)
             hass.config_entries.async_update_entry(entry, data=data)
 
+        hass.async_create_task(
+            check_if_device_has_latest_firmware(hass, device, entry.unique_id)
+        )
```

Comment:
I am against this approach
- This implementation breaks the definition of "local" without the user being able to control it. When users will discover we access Shelly cloud without their permission it will start making noise
- This create a request per device and on every init, so if there is an error and we retry or reload the device there is a new request to Shelly cloud which seems too much just to detect a device which may be with outdated firmware and is not configured correctly so the user is unaware of it.
- Many Shelly users block internet access to their devices by intention, we now create an issue but there isn't a real issue if their firmware is above the minimum we require.

Reply:
I agree it should not be by default, but should be enabled in config per device. Thus allowing a easy way for users that have issues with the device checking internet and didn't block it intentionally.

We briefly discussed this in Teams an nobody were against ;-)